index.js

```diff
@@ -1,68 +1,62 @@
 // import express (after npm install express)
-const express = require('express');
+const express = require('express')
 
 // create new express app and save it as "app"
-const app = express();
+const app = express()
 
-//import package.json
-const pack = require('./package');
+// import package.json
+const packageInfo = require('./package')
 
-//import categories.json
-const all_categories = require('./src/data/categories.json').categories;
-
-//import products.json
-const all_products = require('./src/data/products').products;
+// import dataService.js
+const dataService = require('./dataService.js')
 
 // server configuration
-const PORT = 8080;
+const PORT = 8080
 
 // create a route for the app
 app.get('/', (req, res) => {
-  res.send('Hello World');
-});
-
-function construct_master(){
-	var master_products = all_products;
-	master_products.forEach(
-		(product) => product.categoryName = all_categories.filter(
-		// check if id of category element matches product categoryId
-			each_category => each_category.id === product.categoryId)[0].categoryName);
-	return master_products;
+  res.send('Hello World')
+})
+
+/*
+ * Filter out products of specific category
+ */
+function getProductsByCategory (ctyId) {
+  // Convert the map to an Array, filter out required products and create object from the filtered arra
+  return Object.fromEntries(
+    Array.from(dataService.getCombinedProductMap()).filter(
```

Comment:
Did you check for Object.values or Object.keys ?

Reply:
@riskycase You need to loop through `Object.values(dataService.getCombinedProductMap())` . This is simpler than converting to array.

Reply:
@f4ww4z `Object.values(dataService.getCombinedProductMap())` returns an empty array. The method currently being used is the shortest way to get the products in a specific category since we're dealing with a map